targets/TARGET_STM/TARGET_STM32F0/TARGET_STM32F091xx/device/TOOLCHAIN_ARM_STD/stm32f091xx.sct

```diff
@@ -34,10 +34,16 @@
 
 #define Stack_Size MBED_BOOT_STACK_SIZE
 
-; STM32F091RC: 256KB FLASH (0x40000) + 32KB RAM (0x8000)
+#if defined(STM32F091xC)
+  #define Flash_Size 0x40000
+#else
+  #define Flash_Size 0x20000
+#fi
 
- LR_IROM1 0x08000000 0x40000  {    ; load region size_region
-  ER_IROM1 0x08000000 0x40000  {  ; load address = execution address 
+; STM32F091XX: 128-256KB FLASH (0x20000-0x40000) + 32KB RAM (0x8000)
```

Comment:
Use MBED_ROM_SIZE
https://os.mbed.com/docs/mbed-os/v5.15/porting/porting-bootstrap.html

For NUCLEO_F091RC, in targets.json default configuration, device_name is set to "STM32F091RCTx"
  and then mbed tools sets -DMBED_ROM_SIZE=0x40000

For your custom board, set device_name is set to "STM32F091CBTx,
  you should get -DMBED_ROM_SIZE=0x20000